python/app/services/product_service.py

```diff
@@ -65,3 +65,42 @@ def delete_product(product_id: int) -> bool:
         raise ValueError(f"Failed to delete product: {str(e)}")
     finally:
         db.close()
+
+def update_product(product_id: int, name: str, description: str, categoryId: int, image: Optional[str] = None):
+    """Update an existing product"""
+    db = SessionLocal()
+    try:
+        # Check if product exists
+        product = db.query(ProductModel).get(product_id)
+        if not product:
+            raise ValueError(f"Product with ID {product_id} does not exist")
+            
+        # Validate required fields
+        if not name or name.strip() == "":
+            raise ValueError("Product name cannot be empty")
+        if not description or description.strip() == "":
```

Comment:
Is description a mandatory field in the model? If not then we have to remove this check. 

If its mandatory then ignore this comment